src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -165,6 +167,11 @@ private AuthenticatedEncryptionResult EncryptWithAesCbc(byte[] plaintext, byte[]
         private byte[] DecryptWithAesCbc(byte[] ciphertext, byte[] authenticatedData, byte[] iv, byte[] authenticationTag)
         {
             // Verify authentication Tag
+            if (ShouldValidateAuthenticationTagLength()
+                && SymmetricSignatureProvider.ExpectedSignatureSizeInBytes.TryGetValue(Algorithm, out int expectedTagLength)
```

Comment:
Should the other two checks be added to the ShouldValidateAuthenticationTagLength method since they are necessary for making that decision?

Reply:
I appreciate your comment! The purpose of `ShouldValidateAuthenticationTagLength()` is to check if we need to validate the authentication tag length, not validate it, and would prefer to keep single responsibility for this method 😄